src/prefect/environments/execution/k8s/job.py

```diff
@@ -0,0 +1,212 @@
+import uuid
+from typing import Any, List
+
+import cloudpickle
+import yaml
+
+import prefect
+from prefect.environments.execution import Environment
+from prefect.environments.storage import Docker
+
+
+class KubernetesJobEnvironment(Environment):
+    """
+    KubernetesJobEnvironment is an environment which deploys your flow (stored in a Docker image)
+    as a Kubernetes job. This environment allows (and requires) a custom job YAML spec to be
+    provided.
+
+    When providing a custom YAML job spec the first container in the spec must be the
+    container that the flow runner will be executed on.
+
+    These environment variables are required for cloud do not need to be included because
+    they are instead automatically added and populated during execution:
+    PREFECT__CLOUD__GRAPHQL, PREFECT__CLOUD__AUTH_TOKEN, PREFECT__CONTEXT__FLOW_RUN_ID,
+    PREFECT__CONTEXT__NAMESPACE, PREFECT__CONTEXT__IMAGE, PREFECT__CONTEXT__FLOW_FILE_PATH,
+    PREFECT__CLOUD__USE_LOCAL_SECRETS, PREFECT__ENGINE__FLOW_RUNNER__DEFAULT_CLASS,
+    PREFECT__ENGINE__TASK_RUNNER__DEFAULT_CLASS, PREFECT__LOGGING__LOG_TO_CLOUD
+
+    Additionally, the following command will be applied to the first container:
+    $ /bin/sh -c 'python -c "from prefect.environments import KubernetesJobEnvironment; KubernetesJobEnvironment().run_flow()"'
+
+    Args:
+        - job_spec_file (str, optional): Path to a job spec YAML file
+        - labels (List[str], optional): a list of labels, which are arbitrary string identifiers used by Prefect
+            Agents when polling for work
+    """
+
+    def __init__(self, job_spec_file: str = None, labels: List[str] = None) -> None:
+        self.identifier_label = str(uuid.uuid4())
+        self.job_spec_file = job_spec_file
```

Comment:
You _might_ want to convert this to an absolute path before attaching it to the class; you can use `os.path.abspath(job_spec_file)`

Reply:
@cicdw Should that also be updated to an absolute path on the `DaskKubernetesEnvironment`?

Reply:
Yea, I don't think it's a _huge_ deal but I think if you're storing a file path it's always nicer to see the absolute path in case you change directories or anything.